llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1891,10 +1908,14 @@ class VPWidenIntOrFpInductionRecipe : public VPWidenInductionRecipe {
   VPValue *getVFValue() { return getOperand(2); }
   const VPValue *getVFValue() const { return getOperand(2); }
 
+  VPValue *getStepVector() { return getOperand(3); }
+  const VPValue *getStepVector() const { return getOperand(3); }
+  void setStepVector(VPValue *V) { setOperand(3, V); }
```

Comment:
Can we assert that we only set if it originally is `poison`?

Reply:
Good idea, done in 675992214618dc4365f88a8fa3a1a43072ba4ae4. I've also asserted that the new value is a VPInstruction::StepVector